docs/src/main/sphinx/functions/json.rst

```diff
@@ -6,12 +6,21 @@ Cast to JSON
 ------------
 
 Casting from ``BOOLEAN``, ``TINYINT``, ``SMALLINT``, ``INTEGER``,
-``BIGINT``, ``REAL``, ``DOUBLE`` or ``VARCHAR`` is supported.
-Casting from ``ARRAY``, ``MAP`` or ``ROW`` is supported when the element type of
-the array is one of the supported types, or when the key type of the map
+``BIGINT``, ``REAL``, ``DOUBLE``, or ``VARCHAR`` to JSON is supported.
+
+Casting from ``ARRAY``, ``MAP``, or ``ROW`` is supported when the element type of
+the array is one of the supported types, when the key type of the map
 is ``VARCHAR`` and value type of the map is one of the supported types,
 or when every field type of the row is one of the supported types.
-Behaviors of the casts are shown with the examples below::
+
+.. note::
+
+    Cast operations with supported:ref:`character string types
+    <string-data-types>` treat the input as a string, not validated as JSON.
+    This means that a cast operation with a string-type input of invalid JSON
+    results in a succesful cast to invalid JSON.
```

Comment:
Seems like a bug we should be able to fix. Especially since other CASTs (e.g. string -> number) fail with such invalid inputs and hence the JSON cast behaviour is inconsistent with others. cc: @martint WDYT?

BTW it would help to mention some alternative here if the user wants to "create" JSON. e.g. `json_parse`.

Reply:
@hashhar not a bug (i mean, hm, ... confusing design), but yes, we should link to `json_parse` here.

Reply:
If I recall correctly, this is just a performance optimization. The cast is effectively a no op -- the representation of a JSON type is still a string internally. If the case does validation, then any operation on the resulting JSON value (json_extract, etc) will have to parse the double twice: once for the cast and once for accessing some part of the value.